apps/condo/domains/user/schema/CheckUserExistenceService.test.js

```diff
@@ -0,0 +1,265 @@
+                    isPhoneVerified: true,
+                    phone: userAttrs.phone,
+                })
+                const [result] = await checkUserExistenceByTestClient(anonymousClient, {
+                    confirmActionToken: confirmPhoneAction.token,
+                    userType: userType,
+                })
+                expect(result).toEqual(expectedResult)
+            })
+
+            test('if user is not exist', async () => {
+                const [confirmPhoneAction] = await createTestConfirmPhoneAction(adminClient, { isPhoneVerified: true })
+                const [result] = await checkUserExistenceByTestClient(anonymousClient, {
+                    confirmActionToken: confirmPhoneAction.token,
+                    userType: userType,
+                })
+                expect(result).toEqual({
+                    userIsExist: false,
+                    nameIsSet: false,
+                    emailIsSet: false,
+                    phoneIsSet: false,
+                    passwordIsSet: false,
+                })
+            })
+
+            test('if user was reset', async () => {
+                const [resetUser, resetUserAttrs] = await createTestUser(adminClient, {
+                    type: userType,
+                })
+                await resetUserByTestClient(adminClient, { user: { id: resetUser.id } })
+                const [confirmPhoneAction] = await createTestConfirmPhoneAction(adminClient, {
+                    isPhoneVerified: true,
+                    phone: resetUserAttrs.phone,
+                })
+                const [result] = await checkUserExistenceByTestClient(anonymousClient, {
+                    confirmActionToken: confirmPhoneAction.token,
+                    userType: resetUserAttrs.type,
+                })
+                expect(result).toEqual({
+                    userIsExist: false,
+                    nameIsSet: false,
+                    emailIsSet: false,
+                    phoneIsSet: false,
+                    passwordIsSet: false,
+                })
+            })
+
+            test('if user was soft deleted', async () => {
+                const [deletedUser, deletedUserAttrs] = await createTestUser(adminClient, {
+                    type: userType,
+                })
+                await updateTestUser(adminClient, deletedUser.id, {
+                    deletedAt: new Date().toDateString(),
+                })
+                const [confirmPhoneAction] = await createTestConfirmPhoneAction(adminClient, {
+                    isPhoneVerified: true,
+                    phone: deletedUserAttrs.phone,
+                })
+                const [result] = await checkUserExistenceByTestClient(anonymousClient, {
+                    confirmActionToken: confirmPhoneAction.token,
+                    userType: deletedUserAttrs.type,
+                })
+                expect(result).toEqual({
+                    userIsExist: false,
```

Comment:
but you woulrnt be able to register when user was soft deleted, will you not?

Reply:
Yes, a user who was soft deleted cannot register again